backend/internal/authorization/openfga.go

```diff
@@ -53,76 +68,134 @@ func NewOpenFGAAuthorizer() (gin.HandlerFunc, error) {
 }
 
 func (o *OpenFGAAuthorizer) Authorize(c *gin.Context) {
-	c.AbortWithStatusJSON(501, gin.H{"error": "OpenFGA authorizer not implemented yet"})
-
-	//token := c.GetHeader("Authorization")
-	//if token == "" {
-	//	c.AbortWithStatusJSON(401, gin.H{"error": "missing authorization token"})
-	//	return
-	//}
-	//
-	//parsedToken, err := parseJWT(token)
-	//if err != nil {
-	//	c.AbortWithStatusJSON(401, gin.H{"error": "invalid token"})
-	//	return
-	//}
-	//
-	//user := parsedToken["sub"].(string)
-	//if user == "" {
-	//	c.AbortWithStatusJSON(401, gin.H{"error": "invalid sub claim"})
-	//	return
-	//}
-	//object := c.FullPath() // Use route path
-	//relation := "access"
-	//
-	//contextualTuples := extractContextualTuplesFromToken(token)
-	//
-	//allowed, err := o.check(user, object, relation, contextualTuples)
-	//if err != nil {
-	//	c.AbortWithStatusJSON(500, gin.H{"error": "unexpected error during authorization check"})
-	//	return
-	//}
-	//if !allowed {
-	//	c.AbortWithStatusJSON(403, gin.H{"error": "forbidden"})
-	//	return
-	//}
-	//
-	//c.Next()
+	token := c.GetHeader("Authorization")
+	if token == "" {
+		c.AbortWithStatusJSON(401, gin.H{"error": "missing authorization token"})
```

Comment:
Usually it's better not to give hint about the reason why authorization/authentication fails